src/live_restore/live_restore_fs.c

```diff
@@ -809,13 +804,61 @@ __live_restore_fh_find_holes_in_dest_file(
     return (ret);
 }
 
+/*
+ * __live_restore_handle_smaller_source --
+ *     If the source file is smaller than the destination file make sure the extent list doesn't
+ *     contain holes past the end of the source file.
+ */
+static int
+__live_restore_handle_smaller_source(WT_SESSION_IMPL *session, WT_LIVE_RESTORE_FS *lr_fs,
```

Comment:
If we expect no sequence of reasonable WiredTiger operations should result in this case happening is it better to assert?

Reply:
Yeah good point. Update in [36be756](https://github.com/wiredtiger/wiredtiger/pull/11332/commits/36be75698a9e569232d714fc62e14dc3ec4209e3). 
The new code returns `EINVAL` from `file_open` instead of asserting, mostly because we can test this behaviour in catch2. I'll think a bit on the benefits of asserting versus and error code.